src/checkbox/checkbox.js

```diff
@@ -15,255 +15,182 @@
 
-  /**
-   * Store strings for class names defined by this component that are used in
-   * JavaScript. This allows us to simply change it in one place should we
-   * decide to modify at a later date.
-   *
-   * @enum {string}
-   * @private
-   */
-  MaterialCheckbox.prototype.CssClasses_ = {
-    INPUT: 'mdl-checkbox__input',
-    BOX_OUTLINE: 'mdl-checkbox__box-outline',
-    FOCUS_HELPER: 'mdl-checkbox__focus-helper',
-    TICK_OUTLINE: 'mdl-checkbox__tick-outline',
-    RIPPLE_EFFECT: 'mdl-js-ripple-effect',
-    RIPPLE_IGNORE_EVENTS: 'mdl-js-ripple-effect--ignore-events',
-    RIPPLE_CONTAINER: 'mdl-checkbox__ripple-container',
-    RIPPLE_CENTER: 'mdl-ripple--center',
-    RIPPLE: 'mdl-ripple',
-    IS_FOCUSED: 'is-focused',
-    IS_DISABLED: 'is-disabled',
-    IS_CHECKED: 'is-checked',
-    IS_UPGRADED: 'is-upgraded'
-  };
+    // Finalize initialization.
+    this.init_();
+  }
 
   /**
-   * Handle change of state.
+   * String constants used in this component.
    *
-   * @param {Event} event The event that fired.
-   * @private
+   * @protected
+   * @return {Object<string, string>}
    */
-  MaterialCheckbox.prototype.onChange_ = function(event) {
-    this.updateClasses_();
-  };
+  static get strings_() {
+    return {
+      CLASS_NAME: 'MaterialCheckbox'
+    };
+  }
 
   /**
-   * Handle focus of element.
+   * CSS classes used in this component.
    *
-   * @param {Event} event The event that fired.
-   * @private
+   * @protected
+   * @return {Object<string, string>}
    */
-  MaterialCheckbox.prototype.onFocus_ = function(event) {
-    this.element_.classList.add(this.CssClasses_.IS_FOCUSED);
-  };
+  static get classes_() {
+    return {
+      ROOT: 'mdl-checkbox',
+      JS: 'mdl-js-checkbox',
+      INPUT: 'mdl-checkbox__input',
 
-  /**
-   * Handle lost focus of element.
-   *
-   * @param {Event} event The event that fired.
-   * @private
-   */
-  MaterialCheckbox.prototype.onBlur_ = function(event) {
-    this.element_.classList.remove(this.CssClasses_.IS_FOCUSED);
-  };
+      IS_FOCUSED: 'is-focused',
+      IS_DISABLED: 'is-disabled',
+      IS_CHECKED: 'is-checked',
+      IS_UPGRADED: 'is-upgraded'
+    };
+  }
 
   /**
-   * Handle mouseup.
+   * Attach all listeners to the DOM.
    *
-   * @param {Event} event The event that fired.
-   * @private
+   * @export
    */
-  MaterialCheckbox.prototype.onMouseUp_ = function(event) {
-    this.blur_();
-  };
+  addEventListeners() {
+    this.input_.addEventListener('change', this.changeListener_);
+    this.input_.addEventListener('focus', this.focusListener_);
+    this.input_.addEventListener('blur', this.blurListener_);
+    this.root_.addEventListener('mouseup', this.mouseUpListener_);
+  }
 
   /**
-   * Handle class updates.
+   * Remove all listeners from the DOM.
    *
-   * @private
+   * @export
    */
-  MaterialCheckbox.prototype.updateClasses_ = function() {
-    this.checkDisabled();
-    this.checkToggleState();
-  };
+  removeEventListeners() {
```

Comment:
Yes, assuming the usual path of "remove component from the DOM and everything goes", you're right (and we'll notice we're not actually calling this method from anywhere).

However, after discussing with @paullewis, we decided it was a good idea to have a programmatic way of removing the event handlers from components while keeping the DOM intact. This would be useful if a user is doing upgrade-downgrade cycles for some reason, or wants to render a component inert temporarily.